sphere-models/src/it/java/io/sphere/sdk/customers/commands/CustomerPasswordResetCommandTest.java

```diff
@@ -1,26 +1,46 @@
 package io.sphere.sdk.customers.commands;
 
+import io.sphere.sdk.client.ClientErrorException;
 import io.sphere.sdk.customers.Customer;
 import io.sphere.sdk.customers.CustomerSignInResult;
 import io.sphere.sdk.customers.CustomerToken;
+import io.sphere.sdk.customers.queries.CustomerByPasswordTokenGet;
 import io.sphere.sdk.test.IntegrationTest;
 import org.junit.Test;
 
 import static io.sphere.sdk.customers.CustomerFixtures.withCustomer;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
 
 public class CustomerPasswordResetCommandTest extends IntegrationTest {
     @Test
     public void execution() throws Exception {
         withCustomer(client(), customer -> {
-            final CustomerToken token = execute(CustomerCreatePasswordTokenCommand.of(customer.getEmail()));
+            final String email = customer.getEmail();
+            final CustomerToken token = execute(CustomerCreatePasswordTokenCommand.of(email));
+            final String tokenValue = token.getValue();//this may need to be sent by email to the customer
+
+            final Customer customerByToken = execute(CustomerByPasswordTokenGet.of(tokenValue));
             final String newPassword = "newPassword";
-            final Customer updatedCustomer = execute(CustomerPasswordResetCommand.of(customer, token, newPassword));
+            final Customer updatedCustomer =
+                    execute(CustomerPasswordResetCommand.of(customerByToken, tokenValue, newPassword));
 
-            final CustomerSignInResult signInResult = execute(CustomerSignInCommand.of(updatedCustomer.getEmail(), newPassword, null));
+            final CustomerSignInResult signInResult = execute(CustomerSignInCommand.of(email, newPassword));
             assertThat(signInResult.getCustomer().getId())
-                    .overridingErrorMessage("customer can sign in with the new password")
+                    .describedAs("customer can sign in with the new password")
```

Comment:
Hoping to see the differences between `describedAs`, `overridingErrorMessage` and `as` in your presentation (I could also check it now, but what's the fun of it? ;))